src/validation.cpp

```diff
@@ -259,12 +259,12 @@ namespace {
 
 CBlockIndex* FindForkInGlobalIndex(const CChain& chain, const CBlockLocator& locator)
 {
+    LOCK(cs_main);
```

Comment:
Looks like you're just adding this for CreateWalletFromFile, which already needs cs_main (and is missing it there). This should be an AssertLockHeld.

Reply:
And why does `CreateWalletFromFile` needs `cs_main`?

Reply:
Hmm, there was some discussion of this in some other PR recently, maybe a @practicalswift PR? In any case, there are a number of things in CreateWalletFromFile that need cs_main (though possibly we want to avoid it and just take it for parts to avoid cs_main during the rescan after #11281), eg chainActive.GetLocator(), chainActive.Genesis (maybe not technically, but probably), FindForkInGlobalIndex, chainActive.Tip(), etc. So definitely worth just taking cs_main there and making this an AssertLockHeld.